main.tf

```
@@ -1,6 +1,7 @@
 locals {
   port            = "${var.port == "" ? "${var.engine == "aurora-postgresql" ? "5432" : "3306"}" : var.port}"
   master_password = "${var.password == "" ? random_id.master_password.b64 : var.password}"
+  sg_count        = "${length(var.allowed_security_groups)}"
```

Unfortunately this won't always work. That's why it's implemented with `allowed_security_groups_count` in this module and many others. Moving the count to a local doesn't always work either.

Reference: https://github.com/hashicorp/terraform/issues/12570

---

@max-rocket-internet thanks for pointing that out. does it mean that the implementation of "count" in the current "aws_subnet" resources of [terraform-aws-vpc module](https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/master/main.tf )(1.64.0) is flawed?

maybe we can just add ingress in aws_security_group 
```suggestion
resource "aws_security_group" "this" {
  name_prefix = "${var.name}-"
  vpc_id      = "${var.vpc_id}"

  tags = "${var.tags}"

  ingress {
    from_port       = "${local.port}"
    to_port         = "${local.port}"
    protocol        = "tcp"
    security_groups = "${var.allowed_security_groups}"
  }
}
```

and remove
```suggestion
resource "aws_security_group_rule" "default_ingress" {
  count = "${var.allowed_security_groups_count}"

  type                     = "ingress"
  from_port                = "${aws_rds_cluster.this.port}"
  to_port                  = "${aws_rds_cluster.this.port}"
  protocol                 = "tcp"
  source_security_group_id = "${element(var.allowed_security_groups, count.index)}"
  security_group_id        = "${aws_security_group.this.id}"
}
```